osutil/disks/size.go

```diff
@@ -0,0 +1,27 @@
+package disks
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+// Size returns the size of the given block device, e.g. /dev/sda1 in
+// bytes as reported by the kernels BLKGETSIZE ioctl.
+func Size(partDevice string) (uint64, error) {
+	fp, err := os.Open(partDevice)
+	if err != nil {
+		return 0, fmt.Errorf("cannot open disk to get size: %v", err)
+	}
+	defer fp.Close()
+
+	// kernel 5.14.8 source says:
+	// #define BLKGETSIZE _IO(0x12,96)	/* return device size /512 (long *arg) */
+	partBlocks, err := unix.IoctlGetInt(int(fp.Fd()), unix.BLKGETSIZE)
```

Comment:
should this be `BLKGETSIZE64` maybe? 

it seems like in blockdev from util-linux there is compatability code for handling cases where that is not defined:

https://github.com/karelzak/util-linux/blob/ac55a4130fcdab12ee6a49f8a8ebd6a17c1890df/include/blkdev.h#L50-L52

but not sure really which one we should be using here...

The other thing we could do which I'm okay with since we already call blockdev other places is just to rely on the blockdev command to do the right thing itself and call `blockdev --getsz`

Reply:
Thanks for the suggestion! I updated the code to use BLKGETSIZE64, I think it's indeed the better choice. I can move to "blockdev --getsz" too, it will be easier to unit test. Feels a bit expensive to run an external program where a ioctl is enough but then this helper is rarely used  (today) so it should be fine either way.

Reply:
I moved this now to `blockdev` because moving to BLKGETSIZE64 triggered a segfault on 32bit systems - golang only offers int unix.IoctlGetInt() which is too "short" for disksize.